python/sglang/bench_one_batch_server.py

```diff
@@ -101,17 +102,23 @@ def run_one_case(
     output_len: int,
     run_name: str,
     result_filename: str,
+    tokenizer,
 ):
-    input_ids = [
-        [int(x) for x in np.random.randint(0, high=16384, size=(input_len,))]
-        for _ in range(batch_size)
-    ]
+    input_requests = sample_random_requests(
+        input_len=input_len,
+        output_len=output_len,
+        num_prompts=batch_size,
+        range_ratio=1.0,
+        tokenizer=tokenizer,
+        dataset_path="",
+        random_sample=True,
+    )
 
     tic = time.time()
     response = requests.post(
         url + "/generate",
         json={
-            "input_ids": input_ids,
+            "text": [text for text, _, _ in input_requests],
```

Comment:
let us still use input_ids here to precisely control the input length. Otherwise, retokenization might introduce some unexpected corner cases.

Reply:
Good point! Will do that later